src/librbd/image/RemoveRequest.cc

```diff
@@ -621,6 +627,24 @@ void RemoveRequest<I>::handle_remove_v1_image(int r) {
   if (!m_old_format) {
     remove_v2_image();
   }
+
+}
+
+template<typename I>
+void RemoveRequest<I>::remove_image_cache() {
```

Comment:
I see, then I'll remove its parent snap cache also when the removed image is the last one who is cloned from this parent on this node.